src/main/kotlin/com/zepben/evolve/cim/iec61970/base/wires/PowerTransformerEnd.kt

```diff
@@ -64,10 +67,104 @@ class PowerTransformerEnd @JvmOverloads constructor(mRID: String = "") : Transfo
     var phaseAngleClock: Int? = null
     var r: Double? = null
     var r0: Double? = null
-    var ratedS: Int? = null
+
+    var ratedS: Int?
+        get() = _sRatings?.firstOrNull()?.ratedS
+        @Deprecated("Use addRating() instead, as this will clear all ratings and is intended for backwards compatibility only",
+            ReplaceWith("addRating(this)"))
+        set(value) {
+            if (value != null) {
+                clearRatings()
+                addRating(value, TransformerCoolingType.UNKNOWN_COOLING_TYPE)
+            } else {
+                clearRatings()
+            }
+
+        }
+
     var ratedU: Int? = null
     var x: Double? = null
     var x0: Double? = null
+    private var _sRatings: MutableList<TransformerEndRatedS>? = null
+
+    /**
+     * The normal apparent power ratings for this transformer by their [TransformerCoolingType],
+     * stored in descending order by ratedS. The largest rating will be at the start.
+     *
+     * The [TransformerEndRatedS.ratedS] attribute shall be a positive value. For a two-winding transformer the values for the high and low voltage sides shall
+     * be identical.
+     *
+     * The returned collection is read only.
+     */
+    val sRatings: List<TransformerEndRatedS> get() = _sRatings.asUnmodifiable()
+
+    fun getRating(coolingType: TransformerCoolingType): TransformerEndRatedS? = _sRatings?.find { it.coolingType == coolingType }
+
+    fun getRating(ratedS: Int): TransformerEndRatedS? = _sRatings?.find { it.ratedS == ratedS }
+
+
+    fun forEachRating(action: BiConsumer<Int, TransformerEndRatedS>) {
+        _sRatings?.forEachIndexed(action::accept)
+    }
+
+    fun numRatings(): Int = _sRatings?.size ?: 0
+
+    /**
+     * Add a normal apparent power rating for this PowerTransformerEnd.
+     * The ratings in the underlying collection will be sorted by [ratedS] in a descending order.
+     *
+     * @param ratedS The normal apparent power rating to set.
+     * @param coolingType The cooling type used for this rating, defaults to [TransformerCoolingType.UNKNOWN_COOLING_TYPE]
+     * @throws IllegalArgumentException if a rating for the provided [coolingType] already exists for this PowerTransformerEnd.
+     */
+    fun addRating(
+        ratedS: Int,
+        coolingType: TransformerCoolingType = TransformerCoolingType.UNKNOWN_COOLING_TYPE,
```

Comment:
Cover default (e.g. addRating(11000) should add rating with `UNKNOWN_COOLING_TYPE`)

Reply:
covered - good catch.